### PR TITLE
Refactor Settings Code, Fix Cyclic Complexity#1621

### LINKED ISSUE #1621: Refactor Settings Code and Fix Cyclic Complexity
**Actual Behaviour**

Refactor Settings Code, fix cyclic complexity to lower than 20.

**Expected Behaviour**

Refactor Settings Code

**Screenshots of the issue**

screenshot 2018-10-07 at 1 25 29 pm

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by fragm3**
Fixes #1621 

Changes: 

- Refactor Settings Code
- Refactored code for separate tabs into different react components
- Used constants file for voiceList

Demo Link: https://pr-1621-fossasia-susi-web-chat.surge.sh

### CODE REVIEW

**Comment by fragm3** [comment]
@akshatnitd @Akshat-Jain @anshumanv please review

File: src/components/ChatApp/Settings/AccountTab.react.js [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+            name: 'timezone',
+          }}
+        />
+      </div>
+    );
+  }
+}
+
+AccountTab.propTypes = {
+  TimeZone: PropTypes.string,
```
Comment (akshatnitd) [inline src/components/ChatApp/Settings/AccountTab.react.js]: Use camelCase for all props

File: src/components/ChatApp/Settings/AccountTab.react.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  UserName: PropTypes.string,
```
Comment (akshatnitd) [inline src/components/ChatApp/Settings/AccountTab.react.js]: Use camelCase for all props

File: src/components/ChatApp/Settings/ChatAppTab.react.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import React, { Component } from 'react';
+import Translate from '../../Translate/Translate.react';
+import Toggle from 'material-ui/Toggle';
+import PropTypes from 'prop-types';
+
+class ChatAppTab extends Component {
```
Comment (akshatnitd) [inline src/components/ChatApp/Settings/ChatAppTab.react.js]: Use functional component here, as we do not need to declare it as a React component

File: src/components/ChatApp/Settings/DevicesTab.react.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
+import React, { Component } from 'react';
+import Translate from '../../Translate/Translate.react';
+import TableComplex from '../../TableComplex/TableComplex.react';
+import SwipeableViews from 'react-swipeable-views';
+import MapContainer from '../../MapContainer/MapContainer.react';
+import PropTypes from 'prop-types';
+
+class DevicesTab extends Component {
```
Comment (akshatnitd) [inline src/components/ChatApp/Settings/DevicesTab.react.js]: Use functional component here, as we do not need to declare it as a React component

File: src/components/ChatApp/Settings/MobileTab.react.js [hunk 5]
```diff
@@ -0,0 +1,164 @@
+import React, { Component } from 'react';
+import Translate from '../../Translate/Translate.react';
+import DropDownMenu from 'material-ui/DropDownMenu';
+import TextField from 'material-ui/TextField';
+import PropTypes from 'prop-types';
+
+class MobileTab extends Component {
+  render() {
```
Comment (akshatnitd) [inline src/components/ChatApp/Settings/MobileTab.react.js]: Use functional component here, as we do not need to declare it as a React component

File: src/components/ChatApp/Settings/MobileTab.react.js [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+            </div>
+          </div>
+        </div>
+      </span>
+    );
+  }
+}
+
+MobileTab.propTypes = {
+  PhoneNo: PropTypes.string,
```
Comment (akshatnitd) [inline src/components/ChatApp/Settings/MobileTab.react.js]: Use camelCase for props

File: src/components/ChatApp/Settings/Settings.react.js [hunk 7]
```diff
@@ -1263,616 +1153,135 @@ class Settings extends Component {
...
+          stopEditing={this.stopEditing}
+          handleChange={this.handleChange}
+          obj={this.state.obj}
+          mapObj={this.state.mapObj}
+          google={this.props.google}
+          centerLat={this.state.centerLat}
+          centerLng={this.state.centerLng}
+          devicenames={this.state.devicenames}
+          rooms={this.state.rooms}
+          macids={this.state.macids}
```
Comment (akshatnitd) [inline src/components/ChatApp/Settings/Settings.react.js]: Camelcase here

File: src/components/ChatApp/Settings/Settings.react.js [hunk 8]
```diff
@@ -1263,616 +1153,135 @@ class Settings extends Component {
...
+          divStyle={divStyle}
+          themeVal={UserPreferencesStore.getTheme()}
+          deviceData={this.state.deviceData}
+          handleRemove={this.handleRemove}
+          handleRemoveConfirmation={this.handleRemoveConfirmation}
+          startEditing={this.startEditing}
+          editIdx={this.state.editIdx}
+          stopEditing={this.stopEditing}
+          handleChange={this.handleChange}
+          obj={this.state.obj}
```
Comment (akshatnitd) [inline src/components/ChatApp/Settings/Settings.react.js]: Use proper naming here

File: src/components/ChatApp/Settings/Settings.react.js [hunk 9]
```diff
@@ -1263,616 +1153,135 @@ class Settings extends Component {
...
-                  value={this.state.PhoneNo}
-                  errorText={this.state.phoneNoError}
-                  floatingLabelText={<Translate text="Phone number" />}
-                />
-              </div>
-            </div>
-          </div>
-        </span>
+        <MobileTab
+          divStyle={divStyle}
```
Comment (akshatnitd) [inline src/components/ChatApp/Settings/Settings.react.js]: Use containerStyle instead

File: src/components/ChatApp/Settings/Settings.react.js [hunk 10]
```diff
@@ -1263,616 +1153,135 @@ class Settings extends Component {
...
-          <Toggle
-            className="settings-toggle"
-            onToggle={this.handleEnterAsSend}
-            labelStyle={{ color: themeForegroundColor }}
-            toggled={this.state.enterAsSend}
-          />
-          <br />
-        </div>
+        <ChatAppTab
+          divStyle={divStyle}
```
Comment (akshatnitd) [inline src/components/ChatApp/Settings/Settings.react.js]: Use containerStyle instead

File: src/components/ChatApp/Settings/ThemeChangeTab.react.js [hunk 11]
```diff
@@ -0,0 +1,89 @@
+import React, { Component } from 'react';
+import Translate from '../../Translate/Translate.react';
+import { RadioButton, RadioButtonGroup } from 'material-ui/RadioButton';
+import RaisedButton from 'material-ui/RaisedButton';
+import ThemeChanger from './ThemeChanger';
+import PropTypes from 'prop-types';
+
+class ThemeChangeTab extends Component {
+  render() {
```
Comment (akshatnitd) [inline src/components/ChatApp/Settings/ThemeChangeTab.react.js]: Use functional component here

File: src/components/ChatApp/Settings/ShareOnSocialMedia.js [hunk 12]
```diff
@@ -3,6 +3,7 @@ import Translate from '../../Translate/Translate.react';
 import FontIcon from 'material-ui/FontIcon';
 import RaisedButton from 'material-ui/RaisedButton';
 import urls from '../../../utils/urls';
+import PropTypes from 'prop-types';
 
 export default class ShareOnSocialMedia extends Component {
   constructor(props) {
```
Comment (akshatnitd) [inline src/components/ChatApp/Settings/ShareOnSocialMedia.js]: Use functional component here

Comment (fragm3) [reply src/components/ChatApp/Settings/ShareOnSocialMedia.js]: Here, component has state, so lets keep it this way only

**Comment by fragm3** [comment]
@akshatnitd done the changes, please review!

**Comment by akshatnitd** [comment]
Thank you for your contribution :+1:

**Pull request merged**